plugins/PostProcessingPlugin/scripts/DisplayInfoOnLCD.py

```diff
@@ -173,251 +232,332 @@ def getSettingDataString(self):
-                lcd_text += "Printing "
-            if not self.getSettingValueByKey("scroll"):
-                lcd_text += "Layer "
-            else:
-                lcd_text += file_name + " - Layer "
-            i = self.getSettingValueByKey("startNum")
-            for layer in data:
-                display_text = lcd_text + str(i)
-                layer_index = data.index(layer)
-                lines = layer.split("\n")
-                for line in lines:
-                    if line.startswith(";LAYER_COUNT:"):
-                        max_layer = line
-                        max_layer = max_layer.split(":")[1]
-                        if self.getSettingValueByKey("startNum") == 0:
-                            max_layer = str(int(max_layer) - 1)
-                    if line.startswith(";LAYER:"):
-                        if self.getSettingValueByKey("maxlayer"):
-                            display_text = display_text + " of " + max_layer
-                            if not self.getSettingValueByKey("scroll"):
-                                display_text = display_text + " " + file_name
+            data = self._display_filename_layer(data)
+        else:
+            data = self._display_progress(data)
+        return data
+
+    # This is from the original 'Display Filename and Layer on LCD'
+    def _display_filename_layer(self, data: str) -> str:
+        data[0] = self._add_stats(data)
+        max_layer = 0
+        lcd_text = "M117 "
+        octo_text = "M118 "
+        if self.getSettingValueByKey("file_name") != "":
+            file_name = self.getSettingValueByKey("file_name")
+        else:
+            file_name = Application.getInstance().getPrintInformation().jobName
+        if self.getSettingValueByKey("addPrefixPrinting"):
+            lcd_text += "Printing "
+            octo_text += "Printing "
+        if not self.getSettingValueByKey("scroll"):
+            lcd_text += "Lay "
+            octo_text += "Layer "
+        else:
+            lcd_text += file_name + " - Layer "
+            octo_text += file_name + " - Layer "
+        i = self.getSettingValueByKey("startNum")
+        for layer in data:
+            display_text = lcd_text + str(i)
+            self.m118_text = octo_text + str(i)
+            layer_index = data.index(layer)
+            lines = layer.split("\n")
+            for line in lines:
+                if line.startswith(";LAYER_COUNT:"):
+                    max_layer = line
+                    max_layer = max_layer.split(":")[1]
+                    if self.getSettingValueByKey("startNum") == 0:
+                        max_layer = str(int(max_layer) - 1)
+                if line.startswith(";LAYER:"):
+                    if self.getSettingValueByKey("maxlayer"):
+                        display_text += "/" + max_layer
+                        self.m118_text += "/" + max_layer
+                        if not self.getSettingValueByKey("scroll"):
```

Comment:
```suggestion
                        if not self.getSettingValueByKey("format_option"):
```
The setting that I could find for this is the format_option, you could change the setting instead of renaming this.

Reply:
This one is also used in multiple checks so if you could, use a similar approach of getting the value once and assigning it to a variable, then using that variable for the checks rather than reading from settings. 

Similar to the way it's handled at the top of the next function.
Mainly in case you're reusing it multiple times within the same function.

Reply:
That is vestigial code from the original "Display File Name".  I would have thrown it out but there may have been people who liked it and used it.
Added "format_option = self.getSettingValueByKey("format_option") so the setting is only retrieved once.